tools/msvs/msi/product.wxs

```diff
@@ -19,10 +19,14 @@
            UpgradeCode="47c07a3a-42ef-4213-a85d-8f5a59077c28">
 
     <Package Languages="!(loc.LocaleId)"
-             InstallerVersion="200"
+             InstallerVersion="500"
              Compressed="yes"
              InstallScope="perMachine"/>
 
+    <Condition Message="This application is only supported on Windows 7, Windows Server 2008 R2, or higher.">
+      <![CDATA[Installed OR (VersionNT >= 601)]]>
```

Comment:
> This works as a negative condition? if this is false then the message is displayed?

Correct. The condition must be true for the installer to continue, otherwise the message is displayed.

I have tested a variety of upgrade scenarios and the current condition works fine. As noted, `Installed` is true when uninstalling and false when installing, *and it refers to the current MSI, not any other installed versions of the product*. So the `Installed` property in the condition is to make sure that this version of the MSI with the condition can always be repaired or removed, even if the rest of the condition became false. The OS version could change due to an OS upgrade, or a service pack removal. This probably is not possible with Windows 7, but it looks like a solid pattern to follow (and it's the recommended way in [Wix documentation](http://wixtoolset.org/documentation/manual/v3/howtos/redistributables_and_install_checks/block_install_on_os.html)).